src/lib-imap/test-imap-date.c

```diff
@@ -25,7 +25,7 @@ static void test_imap_date(void)
 		{ "31-Dec-9999", 253402214400LL },
 #endif
 		/* conversions to maximum values */
-#if TIME_T_MAX_BITS == 31
+#if TIME_T_MAX_BITS == 31 || (TIME_T_MAX_BITS == 32 && defined(TIME_T_SIGNED))
```

Comment:
Hmm. I think these checks are assuming TIME_T_MAX_BITS=31 with signed 32bit time_t. Also m4/gmtime_max.m4 has:

```
      printf "check failed, assuming "
      i_cv_gmtime_max_time_t=31
```

So I'm thinking it might be better to fix the m4 test to output TIME_T_MAX_BITS=31 instead with 32bit signed time_t. Then these other checks wouldn't need to change (except tm_is_too_large() probably).

Reply:
Also in this commented case `TIME_T_MAX_BITS == 31` would never happen, so I guess that should be simply `#if  #if TIME_T_MAX_BITS == 31 || (TIME_T_MAX_BITS == 32 && defined(TIME_T_SIGNED)`